Source/WebCore/platform/text/TextFlags.cpp

```diff
@@ -58,15 +60,69 @@ WTF::TextStream& operator<<(TextStream& ts, Kerning kerning)
     return ts;
 }
 
+template <typename T> using BinOp = std::function<T(T, T)>;
+
+template <typename T>
+T join(const std::vector<T>& vec, const BinOp<T>& binOp)
```

Comment:
This is a classic `reduce()` function. I suggest moving this to `std::vector` right next to `map()`, naming it `reduce()`, and adding a 3rd argument `T initialAccumulator = T()`.

Reply:
You mean moving it to `WTF::Vector` ?

Reply:
Yes